// @chakra-ui/react/dist/cjs/components/steps/index.cjs
"use strict";
'use strict';

var steps = require('./steps.cjs');
var steps$1 = require('@ark-ui/react/steps');
var namespace = require('./namespace.cjs');



exports.StepsCompletedContent = steps.StepsCompletedContent;
exports.StepsContent = steps.StepsContent;
exports.StepsContext = steps.StepsContext;
exports.StepsDescription = steps.StepsDescription;
exports.StepsIndicator = steps.StepsIndicator;
exports.StepsItem = steps.StepsItem;
exports.StepsItemContext = steps.StepsItemContext;
exports.StepsList = steps.StepsList;
exports.StepsNextTrigger = steps.StepsNextTrigger;
exports.StepsNumber = steps.StepsNumber;
exports.StepsPrevTrigger = steps.StepsPrevTrigger;
exports.StepsPropsProvider = steps.StepsPropsProvider;
exports.StepsRoot = steps.StepsRoot;
exports.StepsRootProvider = steps.StepsRootProvider;
exports.StepsSeparator = steps.StepsSeparator;
exports.StepsStatus = steps.StepsStatus;
exports.StepsTitle = steps.StepsTitle;
exports.StepsTrigger = steps.StepsTrigger;
exports.useStepsStyles = steps.useStepsStyles;
Object.defineProperty(exports, "useStepsContext", {
  enumerable: true,
  get: function () { return steps$1.useStepsContext; }
});
exports.Steps = namespace;
