// @chakra-ui/react/dist/cjs/components/hover-card/index.cjs
"use strict";
'use strict';

var hoverCard = require('./hover-card.cjs');
var namespace = require('./namespace.cjs');
var hoverCard$1 = require('@ark-ui/react/hover-card');



exports.HoverCardArrow = hoverCard.HoverCardArrow;
exports.HoverCardArrowTip = hoverCard.HoverCardArrowTip;
exports.HoverCardContent = hoverCard.HoverCardContent;
exports.HoverCardContext = hoverCard.HoverCardContext;
exports.HoverCardPositioner = hoverCard.HoverCardPositioner;
exports.HoverCardPropsProvider = hoverCard.HoverCardPropsProvider;
exports.HoverCardRoot = hoverCard.HoverCardRoot;
exports.HoverCardRootProvider = hoverCard.HoverCardRootProvider;
exports.HoverCardTrigger = hoverCard.HoverCardTrigger;
exports.useHoverCardStyles = hoverCard.useHoverCardStyles;
exports.HoverCard = namespace;
Object.defineProperty(exports, "useHoverCard", {
  enumerable: true,
  get: function () { return hoverCard$1.useHoverCard; }
});
Object.defineProperty(exports, "useHoverCardContext", {
  enumerable: true,
  get: function () { return hoverCard$1.useHoverCardContext; }
});
