// @ark-ui/react/dist/components/tree-view/use-tree-view.js
'use client';
import { useMachine, normalizeProps } from '@zag-js/react';
import * as treeView from '@zag-js/tree-view';
import { useId } from 'react';
import { useEnvironmentContext } from '../../providers/environment/use-environment-context.js';
import { useLocaleContext } from '../../providers/locale/use-locale-context.js';
import { useEvent } from '../../utils/use-event.js';

const useTreeView = (props = {}) => {
  const { getRootNode } = useEnvironmentContext();
  const { dir } = useLocaleContext();
  const initialContext = {
    id: useId(),
    dir,
    getRootNode,
    selectedValue: props.defaultSelectedValue,
    expandedValue: props.defaultExpandedValue,
    ...props
  };
  const context = {
    ...initialContext,
    selectedValue: props.selectedValue,
    expandedValue: props.expandedValue,
    onFocusChange: useEvent(props.onFocusChange, { sync: true }),
    onExpandedChange: useEvent(props.onExpandedChange),
    onSelectionChange: useEvent(props.onSelectionChange)
  };
  const [state, send] = useMachine(treeView.machine(initialContext), { context });
  return treeView.connect(state, send, normalizeProps);
};

export { useTreeView };
