// @ark-ui/react/dist/components/accordion/index.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const accordionContext = require('./accordion-context.cjs');
const accordionItem = require('./accordion-item.cjs');
const accordionItemContent = require('./accordion-item-content.cjs');
const accordionItemContext = require('./accordion-item-context.cjs');
const accordionItemIndicator = require('./accordion-item-indicator.cjs');
const accordionItemTrigger = require('./accordion-item-trigger.cjs');
const accordionRoot = require('./accordion-root.cjs');
const accordionRootProvider = require('./accordion-root-provider.cjs');
const useAccordion = require('./use-accordion.cjs');
const useAccordionContext = require('./use-accordion-context.cjs');
const useAccordionItemContext = require('./use-accordion-item-context.cjs');
const accordion$1 = require('./accordion.cjs');
const accordion = require('@zag-js/accordion');



exports.AccordionContext = accordionContext.AccordionContext;
exports.AccordionItem = accordionItem.AccordionItem;
exports.AccordionItemContent = accordionItemContent.AccordionItemContent;
exports.AccordionItemContext = accordionItemContext.AccordionItemContext;
exports.AccordionItemIndicator = accordionItemIndicator.AccordionItemIndicator;
exports.AccordionItemTrigger = accordionItemTrigger.AccordionItemTrigger;
exports.AccordionRoot = accordionRoot.AccordionRoot;
exports.AccordionRootProvider = accordionRootProvider.AccordionRootProvider;
exports.useAccordion = useAccordion.useAccordion;
exports.useAccordionContext = useAccordionContext.useAccordionContext;
exports.useAccordionItemContext = useAccordionItemContext.useAccordionItemContext;
exports.Accordion = accordion$1;
Object.defineProperty(exports, "accordionAnatomy", {
  enumerable: true,
  get: () => accordion.anatomy
});
