// @chakra-ui/react/dist/cjs/components/rating-group/index.cjs
"use strict";
'use strict';

var ratingGroup = require('./rating-group.cjs');
var ratingGroup$1 = require('@ark-ui/react/rating-group');
var namespace = require('./namespace.cjs');



exports.RatingGroupContext = ratingGroup.RatingGroupContext;
exports.RatingGroupControl = ratingGroup.RatingGroupControl;
exports.RatingGroupHiddenInput = ratingGroup.RatingGroupHiddenInput;
exports.RatingGroupItem = ratingGroup.RatingGroupItem;
exports.RatingGroupItemContext = ratingGroup.RatingGroupItemContext;
exports.RatingGroupItemIndicator = ratingGroup.RatingGroupItemIndicator;
exports.RatingGroupLabel = ratingGroup.RatingGroupLabel;
exports.RatingGroupPropsProvider = ratingGroup.RatingGroupPropsProvider;
exports.RatingGroupRoot = ratingGroup.RatingGroupRoot;
exports.RatingGroupRootProvider = ratingGroup.RatingGroupRootProvider;
exports.useRatingGroupStyles = ratingGroup.useRatingGroupStyles;
Object.defineProperty(exports, "useRatingGroupContext", {
  enumerable: true,
  get: function () { return ratingGroup$1.useRatingGroupContext; }
});
Object.defineProperty(exports, "useRatingGroupItemContext", {
  enumerable: true,
  get: function () { return ratingGroup$1.useRatingGroupItemContext; }
});
exports.RatingGroup = namespace;
