// @chakra-ui/react/dist/cjs/components/popover/index.cjs
"use strict";
'use strict';

var popover = require('./popover.cjs');
var popover$1 = require('@ark-ui/react/popover');
var namespace = require('./namespace.cjs');



exports.PopoverAnchor = popover.PopoverAnchor;
exports.PopoverArrow = popover.PopoverArrow;
exports.PopoverArrowTip = popover.PopoverArrowTip;
exports.PopoverBody = popover.PopoverBody;
exports.PopoverCloseTrigger = popover.PopoverCloseTrigger;
exports.PopoverContent = popover.PopoverContent;
exports.PopoverContext = popover.PopoverContext;
exports.PopoverDescription = popover.PopoverDescription;
exports.PopoverFooter = popover.PopoverFooter;
exports.PopoverHeader = popover.PopoverHeader;
exports.PopoverPositioner = popover.PopoverPositioner;
exports.PopoverPropsProvider = popover.PopoverPropsProvider;
exports.PopoverRoot = popover.PopoverRoot;
exports.PopoverRootProvider = popover.PopoverRootProvider;
exports.PopoverTitle = popover.PopoverTitle;
exports.PopoverTrigger = popover.PopoverTrigger;
exports.usePopoverStyles = popover.usePopoverStyles;
Object.defineProperty(exports, "usePopover", {
  enumerable: true,
  get: function () { return popover$1.usePopover; }
});
Object.defineProperty(exports, "usePopoverContext", {
  enumerable: true,
  get: function () { return popover$1.usePopoverContext; }
});
exports.Popover = namespace;
