// @chakra-ui/react/dist/cjs/components/drawer/index.cjs
"use strict";
'use strict';

var drawer = require('./drawer.cjs');
var dialog = require('@ark-ui/react/dialog');
var namespace = require('./namespace.cjs');



exports.DrawerActionTrigger = drawer.DrawerActionTrigger;
exports.DrawerBackdrop = drawer.DrawerBackdrop;
exports.DrawerBody = drawer.DrawerBody;
exports.DrawerCloseTrigger = drawer.DrawerCloseTrigger;
exports.DrawerContent = drawer.DrawerContent;
exports.DrawerContext = drawer.DrawerContext;
exports.DrawerDescription = drawer.DrawerDescription;
exports.DrawerFooter = drawer.DrawerFooter;
exports.DrawerHeader = drawer.DrawerHeader;
exports.DrawerPositioner = drawer.DrawerPositioner;
exports.DrawerRoot = drawer.DrawerRoot;
exports.DrawerRootPropsProvider = drawer.DrawerRootPropsProvider;
exports.DrawerRootProvider = drawer.DrawerRootProvider;
exports.DrawerTitle = drawer.DrawerTitle;
exports.DrawerTrigger = drawer.DrawerTrigger;
exports.useDrawerStyles = drawer.useDrawerStyles;
Object.defineProperty(exports, "useDrawer", {
  enumerable: true,
  get: function () { return dialog.useDialog; }
});
Object.defineProperty(exports, "useDrawerContext", {
  enumerable: true,
  get: function () { return dialog.useDialogContext; }
});
exports.Drawer = namespace;
