// @chakra-ui/react/dist/cjs/components/clipboard/clipboard.cjs
"use strict";
"use client";
'use strict';

var clipboard = require('@ark-ui/react/clipboard');
var createSlotRecipeContext = require('../../styled-system/create-slot-recipe-context.cjs');

const {
  withProvider,
  withContext,
  useStyles: useClipboardStyles,
  PropsProvider
} = createSlotRecipeContext.createSlotRecipeContext({ key: "clipboard" });
const ClipboardRootProvider = withProvider(clipboard.Clipboard.RootProvider, "root", { forwardAsChild: true });
const ClipboardRoot = withProvider(
  clipboard.Clipboard.Root,
  "root",
  { forwardAsChild: true }
);
const ClipboardPropsProvider = PropsProvider;
const ClipboardTrigger = withContext(clipboard.Clipboard.Trigger, "trigger", { forwardAsChild: true });
const ClipboardControl = withContext(clipboard.Clipboard.Control, "control", { forwardAsChild: true });
const ClipboardIndicator = withContext(clipboard.Clipboard.Indicator, "indicator", { forwardAsChild: true });
const ClipboardInput = withContext(clipboard.Clipboard.Input, "input", { forwardAsChild: true });
const ClipboardLabel = withContext(clipboard.Clipboard.Label, "label", { forwardAsChild: true });
const ClipboardContext = clipboard.Clipboard.Context;

exports.ClipboardContext = ClipboardContext;
exports.ClipboardControl = ClipboardControl;
exports.ClipboardIndicator = ClipboardIndicator;
exports.ClipboardInput = ClipboardInput;
exports.ClipboardLabel = ClipboardLabel;
exports.ClipboardPropsProvider = ClipboardPropsProvider;
exports.ClipboardRoot = ClipboardRoot;
exports.ClipboardRootProvider = ClipboardRootProvider;
exports.ClipboardTrigger = ClipboardTrigger;
exports.useClipboardStyles = useClipboardStyles;
