// @chakra-ui/react/dist/cjs/components/file-upload/index.cjs
"use strict";
'use strict';

var fileUpload = require('./file-upload.cjs');
var fileUpload$1 = require('@ark-ui/react/file-upload');
var namespace = require('./namespace.cjs');



exports.FileUploadClearTrigger = fileUpload.FileUploadClearTrigger;
exports.FileUploadContext = fileUpload.FileUploadContext;
exports.FileUploadDropzone = fileUpload.FileUploadDropzone;
exports.FileUploadDropzoneContent = fileUpload.FileUploadDropzoneContent;
exports.FileUploadItem = fileUpload.FileUploadItem;
exports.FileUploadItemContent = fileUpload.FileUploadItemContent;
exports.FileUploadItemDeleteTrigger = fileUpload.FileUploadItemDeleteTrigger;
exports.FileUploadItemGroup = fileUpload.FileUploadItemGroup;
exports.FileUploadItemName = fileUpload.FileUploadItemName;
exports.FileUploadItemPreview = fileUpload.FileUploadItemPreview;
exports.FileUploadItemPreviewImage = fileUpload.FileUploadItemPreviewImage;
exports.FileUploadItemSizeText = fileUpload.FileUploadItemSizeText;
exports.FileUploadLabel = fileUpload.FileUploadLabel;
exports.FileUploadPropsProvider = fileUpload.FileUploadPropsProvider;
exports.FileUploadRoot = fileUpload.FileUploadRoot;
exports.FileUploadRootProvider = fileUpload.FileUploadRootProvider;
exports.FileUploadTrigger = fileUpload.FileUploadTrigger;
exports.useFileUploadStyles = fileUpload.useFileUploadStyles;
Object.defineProperty(exports, "useFileUpload", {
  enumerable: true,
  get: function () { return fileUpload$1.useFileUpload; }
});
Object.defineProperty(exports, "useFileUploadContext", {
  enumerable: true,
  get: function () { return fileUpload$1.useFileUploadContext; }
});
exports.FileUpload = namespace;
